### DISCUSSION TITLE
[Bug]: Firmware Notification - Position can only be sent once every 5 seconds

### DISCUSSION THREAD

**Discussion by akrabu** [meshtastic/firmware#4875] [Troubleshooting]
### Category

Other

### Hardware

Rak4631, Seeed Card Tracker T1000-E

### Firmware Version

2.5.2.771cb52

### Description

I loaded 2.5.2 on my MuziWorks R1 (RAK4531) and Seeed T1000-E this morning.  Took both to work today to see if the Seeed could replace my R1 as an EDC node.  Both worked fine until an hour before I left for work.  Suddenly I started getting a message about "position can only be sent once every 5 seconds" repeatedly, which was weird.  I was using smart location, with stationary set to 3 hours, and on the move set to 15 minutes - minutes, not seconds.

It started hammering my notifications though, so I turned off smart position in case it was wonky GPS readings causing the issue.  Even with it set to every hour and smart location off, I was still getting the message every 5 seconds so I powered off the node.  Then I switched to my R1 (Wisblock), and it, too, was doing the exact same thing.  I powered it off, and powered my Seeed node on again.  It also seems to happen whether I use the internal GPS or the phone GPS with the Seeed node.

I checked the node metrics out of curiosity, and was shocked to see the battery was dead.  Both nodes are Client Mute, and tuned mostly for battery life.  But something had happened and the voltage very quickly dropped.  Also, temperature readings seemed to start... going back in time, around the same time?  Very odd output.  See attached screenshots.

I tried doing the "wipe" firmware, and re-updating to 2.5.2 - it immediately started spamming me with notifications again.  Going to downgrade to 2.5.0, though I know that's got issues with the T1000-E.  We'll see.  May have to go down to 2.4.3.

What I don't get is... how did I make it through most of the day without any issue at all?  I wonder what triggered the sudden location spamming, huge power drain, and weird temp/time readings?  I was up on Marquam Hill and couldn't  reach my other node.  There's no way it was picking up time from other nodes - because there really aren't any where I was haha

UPDATE: Switching to v.2.5.0 at least stopped the constant warning notifications.

![image0](https://github.com/user-attachments/assets/32fc04d5-9844-4c28-b35e-b17e86e73e1a)

![IMG_4851](https://github.com/user-attachments/assets/51685c84-0f3c-4b00-9f8e-b77f6fa5a7a5)

![IMG_4852](https://github.com/user-attachments/assets/1addd1a9-4d3a-4930-91fb-4d8c45ee3622)

![IMG_4850](https://github.com/user-attachments/assets/bbe76b56-8335-4b2d-9baa-a5b921b9ebed)

![IMG_4854](https://github.com/user-attachments/assets/f2a4e954-ee09-4158-ae8f-84067176ded1)

**Comment by thebentern** [comment]
Position config does not control the app sending positions to the device. That is actually under App Settings. I've witnessed this same behavior from the iOS app after a series of disconnect / reconnect cycles. It seems that timer sending the position in the background gets some zombie instances going and starts spamming consistently until issuing a hard kill of the app and restarting. 

I believe this behavior happens more than people realize, and now it's made visible so that a user can take action and restart the app.

The telemetry metrics seem nose-diving is interesting. I wonder if the spam of position packets going out over lora drained the device battery.

**Reply by akrabu** [reply]
> Position config does not control the app sending positions to the device. That is actually under App Settings. I've witnessed this same behavior from the iOS app after a series of disconnect / reconnect cycles. It seems that timer sending the position in the background gets some zombie instances going and starts spamming consistently until issuing a hard kill of the app and restarting.
> 
> I believe this behavior happens more than people realize, and now it's made visible so that a user can take action and restart the app.
> 
> The telemetry metrics seem nose-diving is interesting. I wonder if the spam of position packets going out over lora drained the device battery.

I did try force-quitting the app, hoping to get the notifications to stop.  They did, but immediately resumed upon opening the app.  In the app settings, I have it set to share the location every 5 minutes.  Also at first I was using the hardware GPS in the Seeed tracker, but only switched (via "disabled") to try to get the messages to stop.

The only thing I could do, other than powering it off, to get it to stop was to downgrade to v2.5.0

I had reported similar "bad location: horizontal activity" logs to the Discord, all the way back in 2.4.0 - but were told that was normal and it just meant that the "smart location" hadn't been triggered because the location wasn't far enough.  Was that incorrect?  Literally all my nodes with smart location do that when I'm connected to them.

**Reply by thebentern** [reply]
I would try changing the interval in App Settings, and if that doesn't work, perhaps a Clear App Data. What's unfortunate is that the app's position spamming will still be happening on your device downgraded to 2.5.0. You just won't be alerted to it. If it happens rapidly enough, I've seen devices completely lock up. From my experience it's always ceased after a hard close of the app though. It's very unusual that the issue is resuming immediately.

**Reply by akrabu** [reply]
> I would try changing the interval in App Settings, and if that doesn't work, perhaps a Clear App Data. What's unfortunate is that the app's position spamming will still be happening on your device downgraded to 2.5.0. You just won't be alerted to it. If it happens rapidly enough, I've seen devices completely lock up. From my experience it's always ceased after a hard close of the app though. It's very unusual that the issue is resuming immediately.

Oh, no, when I was on 2.4.0-2.4.2 and was discussing this issue, I definitely tried that - it always almost immediately started filling the log again.  Just tested it again right now.  Had a few, force quit, counted to 10, reopened, let it sit for a minute... yep there's some more.

Regardless I have it set to 5 minutes.  That should be well within the "once every 5 seconds" it's complaining about, no?

Update: Set it to the maximum of 15 minutes and turned off smart location.  Refreshed, waited a few seconds, refreshed again... there's 32 more of them.  I realize these messages are constant - they have been since I got into the hobby.  But never before have I seen the voltage drop like that.  Also, I checked back... those messages had been going *all day* as they always do.  But only around the time the voltage dropped did I start getting that actual firmware warning notification. 

If it was just that error message in the log... shouldn't I have been getting actual notification all day long?

![IMG_4855](https://github.com/user-attachments/assets/a06f3e2f-974c-4c37-9734-a7b2bb748c54)

**Reply by thebentern** [reply]
> Regardless I have it set to 5 minutes. That should be well within the "once every 5 seconds" it's complaining about, no?

Right, it definitely should be. I'm trying to rule out some weird timer value getting cached in that app setting.

**Reply by akrabu** [reply]
Okay, for a test...  I disabled location sharing from the app.

Refreshed.  Saw more.  Refreshed again, still saw even more.  Force quit the app.  That stopped it.  Perhaps this _is_ an iOS bug.

I'm going to load 2.5.2 back on, keep app location sharing off, and use the internal GPS and see what happens.  

Ps. I do see THIS error, which I never noticed before - probably because my log file was too full of those location errors.  Any idea what this is about, since I've got you here? haha

![IMG_4860](https://github.com/user-attachments/assets/99c653dd-54c8-4f0f-b2a3-218eac9aba69)

**Reply by akrabu** [reply]
@thebentern What I'm finding on the Discord, looking for this issue, is that those errors are from the "smart position" setting in the Meshtastic IOS app settings (not the device settings).  It's function is to not send old or inaccurate positions to the device - and therefore should not be broadcasting them.

https://github.com/meshtastic/Meshtastic-Apple/blob/e00ffe6214cb894511e2762fee351aca0539ec48/Meshtastic/Helpers/LocationsHandler.swift#L85

https://discord.com/channels/867578229534359593/871553024030556250/1269141021548216382

https://discord.com/channels/867578229534359593/871553024030556250/1269152768283902033

So... in theory... these errors in the log are positions being DISCARDED, and NOT transmitted...  so why should it be sending a location, let alone more than once every 5 seconds, to trigger a firmware notification?

**Reply by thebentern** [reply]
> So... in theory... these errors in the log are positions being DISCARDED, and NOT transmitted

That makes sense. If you want to see what's happening on the device, you can enable Debug Logs under Security config. It will show essentially the same thing as the serial logging over bluetooth in the Debug Logs view at the bottom of Settings.
![image](https://github.com/user-attachments/assets/d4c5fb7e-a8f8-4e54-989f-eead3d852151)

**Comment by fifieldt** [comment]
This was a bug which was fixed :)
